### PR TITLE
DEV: Keithley 2700 class and one-off for IM3L0

### PR DESCRIPTION
**Pull request by KaushikMalapati**
## Description

Removed IM3L0 detailed screen and added a K2700 (Keithley) class and one-off instance for the IM3L0 Keithley which uses a custom pydm screen. Also added a K2700 component to IM3L0.

## Motivation and Context

Requested here https://jira.slac.stanford.edu/browse/ECS-3778. Adds a Keithley pydm screen to the IM3L0's detailed screen in a convenient location above im3l0_power_meter.

## How Has This Been Tested?

This has been tested by using my local fork of pcdsdevices with typhos to see if the screen for IM3L0 loads correctly - no longer has Keithley screen embedded at top with custom detailed template, has K2700 component screen, components are correctly reordered.

## Where Has This Been Documented?

New upcoming_release_notes: 1169-Keithley_2700.rst

image

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [ ] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by KaushikMalapati** [comment]
Is this error from my end? It looks like documentation building is trying to import IPython but never installed that module.

**Comment by klauer** [comment]
> Is this error from my end? It looks like documentation building is trying to import IPython but never installed that module. I also tried re-running that job on a previous pr that was approved, #1162, and it failed with the same error (though it passed last time).

Not your fault. Let me take a look.

**Comment by klauer** [comment]
The docs building should be resolved; just need to update your branch to merge in master's changes here

File: pcdsdevices/pim.py [hunk 1]
```diff
@@ -559,17 +559,48 @@ class IM2K0(LCLS2ImagerBase):
...
     """
     pass
 
 
+@reorder_components(
+    end_with=['k2700', 'power_meter', 'yag_thermocouple', 'led']
+)
+class IM3L0(PPM):
+    """
+    One-off subclass of PPM to include this device's Keithely 2700
```
Comment (klauer) [inline pcdsdevices/pim.py:595]:
```suggestion
    One-off subclass of PPM to include this device's Keithley 2700
```

File: pcdsdevices/pim.py [hunk 2]
```diff
@@ -559,17 +559,48 @@ class IM2K0(LCLS2ImagerBase):
...
+    Keithley 2700 digital multimeter.
+
+    Currently supports reading voltage and current, direct and
+    alternating, but can be extended to measure other quantities
+    (resistance, temperature), have configurable range and integration
+    time, and allow for remote control of a K2700.
+    """
+    idn = Cpt(EpicsSignalRO, ":Identity", kind="normal")
+    reading = Cpt(EpicsSignalRO, ":Reading", kind="normal")
+    dcv = Cpt(EpicsSignalRO, ":GetDCV", kind="normal")
```
Comment (klauer) [inline pcdsdevices/pim.py:573]:
```suggestion
    dcv = Cpt(EpicsSignalRO, ":GetDCV", kind="normal", doc="Some description here")
```
Would you mind adding some descriptions to these? While I know what IDN/DCV/ACV/DCI/ACI are, others might not. What goes in "Reading" isn't obvious to me, though

File: pcdsdevices/pim.py [hunk 3]
```diff
@@ -559,17 +559,48 @@ class IM2K0(LCLS2ImagerBase):
...
+    end_with=['k2700', 'power_meter', 'yag_thermocouple', 'led']
+)
+class IM3L0(PPM):
+    """
+    One-off subclass of PPM to include this device's Keithely 2700
+
+    Includes a Keithley 2700 digital multimeter on top of the PPM class, mainly
+    so it shows up on this device's detailed screen.
+    """
+    k2700 = Cpt(IM3L0_K2700, ':SPM:K2700', kind='normal',
```
Comment (klauer) [inline pcdsdevices/pim.py:600]: Sub-devices don't need "kinds", just so you're aware. It doesn't technically hurt anything to have this here, but it also has the same effect as leaving it off.

`kind` is more of a "leaf Signal" sort of thing (as it tells the data acquisition routines what's important for plotting and such) ([ref](https://confluence.slac.stanford.edu/display/PCDS/How+to+Create+an+Ophyd+Device#HowtoCreateanOphydDevice-Choosingtheright%22kind%22))
```suggestion
    k2700 = Cpt(IM3L0_K2700, ':SPM:K2700',
```

**Review by klauer — COMMENTED** [review]
I think this looks good overall! A few tweaks and it should be ready to merge.

**Comment by KaushikMalapati** [comment]
Is this error from my end? 

> The docs building should be resolved; just need to update your branch to merge in master's changes here

What was the issue? I assume that this worked without IPython as a requirement before so I'm unsure why only this repo would see this issue if it's using python-standard.yml.

**Comment by klauer** [comment]
> What was the issue? I assume that this worked without IPython as a requirement before so I'm unsure why only this repo would see this issue if it's using python-standard.yml.

We must have been installing Python through a dependency-of-a-dependency is my best guess. Since we're using it directly in the documentation generation, it _should have_ been listed in `docs-requirements.txt` directly. So the test suite failure helped us fix a packaging bug - a good thing overall.

**Review by klauer — APPROVED** [review]

**Comment by KaushikMalapati** [comment]
> > What was the issue? I assume that this worked without IPython as a requirement before so I'm unsure why only this repo would see this issue if it's using python-standard.yml.
> 
> We must have been installing Python through a dependency-of-a-dependency is my best guess. Since we're using it directly in the documentation generation, it _should have_ been listed in `docs-requirements.txt` directly. So the test suite failure helped us fix a packaging bug - a good thing overall.

Wouldn't any other repo that uses pcds-ci-helpers/.github/workflows/python-docs.yml (python-standard) also need to add IPython as an requirement? Feels like something that should be installed in that repo.

**Comment by klauer** [comment]
> Wouldn't any other repo that uses pcds-ci-helpers/.github/workflows/python-docs.yml (python-standard) also need to add IPython as an requirement? Feels like something that should be installed in that repo.

If the requirements are specific to continuous integration and are not technically required by the package, they would either go in `pcds-ci-helpers` or in the actions configuration we have in this repository (`.github/workflows/standard.yml`)
However, we might want to build docs locally - in which case it would be inconvenient for the requirements to be solely listed in an unrelated repository. That's why we have them in `docs-requirements.txt` here

**Pull request merged**